core/core-impl/src/main/java/ai/timefold/solver/core/impl/score/stream/IntCounter.java

```diff
@@ -0,0 +1,17 @@
+package ai.timefold.solver.core.impl.score.stream;
+
+public final class IntCounter {
+    private int count;
+
+    public void increment() {
+        count++;
+    }
+
+    public void decrement() {
+        count--;
+    }
+
+    public Integer result() {
```

Comment:
Since there are no generics here, can we return `int`? Let's return what we have, and deal with boxing elsewhere.